### PR TITLE
aws-secretsmanager-caching: init at 1.1.1.5

### PR DESCRIPTION
**Pull request by tomaskala**
###### Description of changes

The AWS Secrets Manager Python caching client enables in-process caching of secrets for Python applications. 

I disabled running integration tests, since they require networking. This reduces the test suite size from 39 integration and unit tests to 33 unit tests.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by tomaskala** [comment]
> Please squash the commit `Fix the required Python version` into the package addition commit. Do you mind adding yourself into the maintainers array of the package?

@onny I squashed the commit you wanted. I had already added myself into the maintainers list, though it was in a different commit than the one that introduced the package.

File: pkgs/development/python-modules/aws-secretsmanager-caching/default.nix [hunk 1]
```diff
@@ -49,6 +49,6 @@ buildPythonPackage rec {
       The AWS Secrets Manager Python caching client enables in-process caching of secrets for Python applications.
     '';
     license = licenses.asl20;
-    maintainers = [ ];
+    maintainers = with maintainers; [ tomaskala ];
```
Comment (onny) [inline pkgs/development/python-modules/aws-secretsmanager-caching/default.nix:52]: This change should belong to the commit adding the aws-secretsmanager-caching package

File: pkgs/development/python-modules/aws-secretsmanager-caching/default.nix [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    inherit pname version;
+    sha256 = "5cee2762bb89b72f3e5123feee8e45fbe44ffe163bfca08b28f27b2e2b7772e1";
+  };
+
+  propagatedBuildInputs = [
+    botocore
+  ];
+
+  nativeCheckInputs = [
+    pytest-cov
```
Comment (onny) [inline pkgs/development/python-modules/aws-secretsmanager-caching/default.nix:29]: We usually don't include coverage tests. Please remove `pytest-cov` and propably `codecov` from the dependencies and patch the Python build files to disable coverage tests

File: pkgs/development/python-modules/aws-secretsmanager-caching/default.nix [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    # Integration tests require networking.
+    "test/integ"
+  ];
+
+  pythonImportsCheck = [
+    "aws_secretsmanager_caching"
+  ];
+
+  meta = with lib; {
+    description = "Client-side AWS Secrets Manager caching library";
```
Comment (onny) [inline pkgs/development/python-modules/aws-secretsmanager-caching/default.nix:45]: Should propably be `Client-side AWS secrets manager caching library`

File: maintainers/maintainer-list.nix [hunk 4]
```diff
@@ -15901,6 +15901,12 @@
     githubId = 8577941;
     name = "Kevin Rauscher";
   };
+  tomaskala = {
```
Comment (onny) [inline maintainers/maintainer-list.nix:15904]: Please change the commit message to something like `maintainers: add tomaskala`

**Review by onny — COMMENTED** [review]
Sorry for the small nitpicks, but these changes would be good to have

**Comment by onny** [comment]
Thank you for the fast fix :) Have you checked it builds properly? I get following error with nixpkgs-review:
```
ERROR: Could not find a version that satisfies the requirement setuptools-scm (from versions: none)
ERROR: No matching distribution found for setuptools-scm
```

**Comment by tomaskala** [comment]
> Thank you for the fast fix :) Have you checked it builds properly? I get following error with nixpkgs-review:
> 
> ```
> ERROR: Could not find a version that satisfies the requirement setuptools-scm (from versions: none)
> ERROR: No matching distribution found for setuptools-scm
> ```

Whoops! Sorry about that, I was only building with `nix-build` and forgot I had `sandbox = false` in my Nix config. I was missing `setuptools-scm` in `nativeBuildInputs` and also had to remove `pytest-runner` from `setup.py`.

**Comment by onny** [comment]
Result of `nixpkgs-review pr 231068` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python310Packages.aws-secretsmanager-caching</li>
    <li>python310Packages.aws-secretsmanager-caching.dist</li>
    <li>python311Packages.aws-secretsmanager-caching</li>
    <li>python311Packages.aws-secretsmanager-caching.dist</li>
  </ul>
</details>

**Comment by tomaskala** [comment]
> Thank you for your contribution :)

Thank you for your review :)

**Pull request merged**